module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -308,11 +293,290 @@ public function holdAction()
+                    $result->code = 'OD_CODE_ALREADY_ON_HOLD';
+                }
+            } else {
+                $result->status = true;
+            }
+        }
+        return $result;
+    }
+
+    /**
+     * Checkout Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getCheckoutRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $edition = $this->params()->fromPost(
+            'edition',
+            $this->params()->fromQuery('edition', false)
+        );
+        if ($edition) {
+            $od_id = $edition;
+        }
+        $result = $this->connector->doOverdriveCheckout($od_id);
+        return $result;
+    }
+
+    /**
+     * Place Hold Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getPlaceHoldRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $email = $this->params()->fromPost('email');
+        $result = $this->connector->placeOverDriveHold($od_id, $email);
+        if ($result->status) {
+            $result->code = 'od_hold_place_success';
+            $result->codeParams = ['%%holdListPosition%%' => $result->data->holdListPosition];
+        } else {
+            $result->code = 'od_hold_place_failure';
+        }
+        return $result;
+    }
+
+    /**
+     * Edit Hold Email Confirm Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getEditHoldEmailConfRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $hold = $this->connector->getHold($od_id, true);
+        $result = $this->connector->getResultObject(true);
+        $result->data = (object)[];
+        $result->data->hold = $hold;
+        return $result;
+    }
+
+    /**
+     * Edit Hold Email Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getEditHoldEmailRes()
+    {
+        $od_id = $this->params()->fromQuery('od_id');
+        $email = $this->params()->fromPost('email');
+        $result = $this->connector->updateOverDriveHold($od_id, $email);
+        $result->code = $result->status ? 'od_hold_update_success' : 'od_hold_update_failure';
+        return $result;
+    }
+
+    /**
+     * Return Title Confirmation Result
+     *
+     * Get result of the action
+     *
+     * @return obj Result Object
+     */
+    public function getReturnTitleConfirmResult()
+    {
+        $result = $this->connector->getResultObject();
+        $rec_id = $this->params()->fromQuery('rec_id');
+        // Load the SolrOverdrive driver.
+        $driver = $this->serviceLocator->get(\VuFind\Record\Loader::class)->load(
+            $rec_id
+        );
+        $formats = $driver->getDigitalFormats();
+        $result->data = (current($formats)->id == 'magazine-overdrive') ?: false;
```

Comment:
Here's another instance of the magazine check. Maybe it's worth creating a support method like `isRecordDriverMagazine` to do this work since it's repeated, whether or not the logic itself needs to be adjusted.